src/scripts/mod.rs

```diff
@@ -0,0 +1,309 @@
+//! Scripting engine to run scripts based on tags
+//!
+//!
+use anyhow::{anyhow, Result};
+use serde_derive::{Deserialize, Serialize};
+use std::fs::{self, File};
+use std::io::{self, prelude::*};
+use std::net::IpAddr;
+use std::path::PathBuf;
+use subprocess::{Exec, ExitStatus};
+use text_placeholder::Template;
+
+pub fn init_scripts() -> Vec<ScriptFile> {
+    let scripts_dir_base = match dirs::home_dir() {
+        Some(dir) => dir,
+        None => panic!("Could not infer scripts path."),
+    };
+    let script_paths = find_scripts(scripts_dir_base).unwrap();
+    debug!("Scripts paths \n{:?}", script_paths);
+    let parsed_scripts = parse_scripts(script_paths);
+    debug!("Scripts parsed \n{:?}", parsed_scripts);
+
+    let script_config = ScriptConfig::read_config().unwrap();
+    debug!("Script config \n{:?}", script_config);
+
+    let mut scripts_to_run: Vec<ScriptFile> = Vec::new();
+
+    // Only Scripts that contain all the tags found in ScriptConfig will be selected.
```

Comment:
By tag do you mean stuff like "core" etc? or do you mean stuff like the author field too? If it's strictly tags like "core, http" etc i can see this being a neat lil security feature (an intended bug)

Reply:
The logic only collect those ScriptFiles from the scripts folder which contains all the tags predefined in the script_config file. For example you can give a crazy unique tag for a script and call that in the ScriptConfig file so only that script will run. We can build more logic around script parsing.